### PR TITLE
[MGDOBR-1152] - Update GH actions to use Node.js 16 and deprecated commands

### PR DESCRIPTION
**Pull request by afalhambra**
[MGDOBR-1152](https://issues.redhat.com/browse/MGDOBR-1152)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Comment by afalhambra** [comment]
Blocked by:

- https://github.com/container-tools/kind-action/pull/8
- https://github.com/imranismail/setup-kustomize/pull/273

This PR has been tested using branch https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/tree/update-node16 and draft PR https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1225

**Review by sutaakar — APPROVED** [review]

**Comment by sutaakar** [comment]
@afalhambra I still see some usage of Node 12. Seems to be used by transitive actions.
Can you please revise that? It seems that at least `setup-maven-action` could be updated to use Node 16 compatible transitive actions.

**Comment by afalhambra** [comment]
@sutaakar - I've updated all the GHA. However, this PR is again blocked by the below pending PRs:

- https://github.com/ScaCap/action-surefire-report/pull/93
- https://github.com/actions-cool/check-user-permission/pull/10

File: .github/workflows/Manager-CI.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ on:
   pull_request_target:
     types: [ opened, labeled, unlabeled, synchronize ]
     branches:
-      - 'main'
+      - 'update-node16'
```
Comment (sutaakar) [inline .github/workflows/Manager-CI.yaml:6]: Unintentional leftover?

**Comment by afalhambra** [comment]
> @sutaakar - I've updated all the GHA. However, this PR is again blocked by the below pending PRs:
> 
> * [Update default runtime to Node.js 16 ScaCap/action-surefire-report#93](https://github.com/ScaCap/action-surefire-report/pull/93)
> * [Action needs to be updated to Node16 actions-cool/check-user-permission#10](https://github.com/actions-cool/check-user-permission/pull/10)

Ok, now, only this one is pending to go:

https://github.com/ScaCap/action-surefire-report/pull/93

**Comment by sutaakar** [comment]
Seems the latest release of `action-surefire-report` is a year ago.
Maybe it may have sense to find action which is better maintained....

**Comment by sutaakar** [comment]
Makes me think, do we really need this action? Is anybody using the Surefire summary from CI summary page?
IMHO it doesn't seem to bring much benefit.

**Comment by sutaakar** [comment]
Seems that we also need https://github.com/stCarolas/setup-maven/pull/28

**Review by jiripetrlik — APPROVED** [review]

**Comment by jiripetrlik** [comment]
@afalhambra 
Can we remove "on hold" label?

**Pull request merged**